### PR TITLE
[PORT] Tgui stack menu

### PR DESCRIPTION
**Pull request by KubeRoot**
## About The Pull Request

Ports the following PRs:
- https://github.com/tgstation/tgstation/pull/53657 (Original PR: https://github.com/VOREStation/VOREStation/pull/8998)
- https://github.com/tgstation/tgstation/pull/53902
  - Only stack menu improvements ported

Provides a nicer looking menu for crafting stuff from material stacks in-hand, featuring a search function.

Additional modifications from tg:
- Preserve recipe order, readd spacers
- Make searching apply to subsection entries, for example "beige" finding "beige comfy chair" inside "comfy chairs"
- Enable sorting and expand subsections when searching
- Subsections no longer have a margin, only their contents do

The changes also required refactoring the format in which data is passed over to tgui. Instead of an associative list of `[title] = recipe`, the recipes are passed as a simple list/array of recipes. The recipes include the title as a field, a recipe can also instead be an object `{ spacer = true }` to indicate a spacer instead of a recipe.

### Sample images:

image

image

## Why It's Good For The Game

Better interfaces make the game more pleasant to play. The ability to search in particular makes the menu much nicer to use, and handle lots of entries better.

## Changelog
:cl:
add: Stack menu now uses tgui and has a search bar.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Can we sort it how it used to be sorted rather than alphabetically?

**Comment by KubeRoot** [comment]
> Can we sort it how it used to be sorted rather than alphabetically?

Gonna look into it

EDIT: Done

File: tgui/packages/tgui/interfaces/Stack.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+   && filterRecipes(recipes, searchText)
+    || recipes;
+
+  const height = Math.max(94 + recipes.length * 26, 250);
+
+  return (
+    <Window
+      width={400}
+      height={Math.min(height, 500)}
+      resizable>
```
Comment (BrumalLake) [inline tgui/packages/tgui/interfaces/Stack.js]: you can get rid of resizable, all tgui windows are automatically now

Comment (KubeRoot) [reply tgui/packages/tgui/interfaces/Stack.js]: Good catch

**Pull request merged**